airbyte-bootloader/src/main/java/io/airbyte/bootloader/BootloaderApp.java

```diff
@@ -162,13 +177,43 @@ private void initPersistences() {
 
       configPersistence = DatabaseConfigPersistence.createWithValidation(configDatabase, jsonSecretsProcessor);
     } catch (final IOException e) {
-      e.printStackTrace();
+      LOGGER.error("Unable to initialize persistence.", e);
     }
   }
 
   public static void main(final String[] args) throws Exception {
-    final var bootloader = new BootloaderApp();
-    bootloader.load();
+    final Configs configs = new EnvConfigs();
+    final FeatureFlags featureFlags = new EnvVariableFeatureFlags();
+
+    // Manual configuration that will be replaced by Dependency Injection in the future
+    final DataSource configsDataSource = DataSourceFactory.create(configs.getConfigDatabaseUser(), configs.getConfigDatabasePassword(),
+        DRIVER_CLASS_NAME, configs.getConfigDatabaseUrl());
+    final DataSource jobsDataSource =
+        DataSourceFactory.create(configs.getDatabaseUser(), configs.getDatabasePassword(), DRIVER_CLASS_NAME, configs.getDatabaseUrl());
+
+    try (final DSLContext configsDslContext = DSLContextFactory.create(configsDataSource, SQLDialect.POSTGRES);
+        final DSLContext jobsDslContext = DSLContextFactory.create(configsDataSource, SQLDialect.POSTGRES)) {
+
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+        configsDslContext.close();
+        jobsDslContext.close();
+        closeDataSource(configsDataSource);
+        closeDataSource(jobsDataSource);
+      }));
```

Comment:
@lmossman Correct.  For some reason, the `javax.sql.DataSource` interface does not extend from `Closeable` or `AutoCloseable`.  Therefore, not all implementations will be closeable, nor can we use it in the try-with-resources block if using the interface.  We would need to explicitly use the `HikariDataSource` type, which defeats the purpose here.  It's a bummer, but luckily just a temporary thing until we can let the framework manage these objects for us.

Reply:
It is worth leaving comments here to explain this in each of the main app? Or perhaps centralising this method into one of the factory classes to make it clear why this is needed? Or are we going to remove this relatively soon so this isn't worth the additional refactor?

Reply:
It can never hurt to have additional comments/context, especially if the code is temporary.  I will update the PR with comments as to why the runtime hook is there.  Let me see how easy it would be generalize the hook while I am at it.

Reply:
@davinchia @lmossman PR has been updated to centralize the shutdown logic (and with comments).  The new class is now in airbyte-commons (`CloseableShutdownHook`).  You can look at any of the `*App` class to see usage.  I will update the Cloud PR to use this class as well (with comment).

Reply:
looks good! exactly what I was thinking of :)